src/ApiConnectors/BaseApiConnector.php

```diff
@@ -47,6 +47,11 @@ public function __construct(AuthenticatedConnection $connection)
         $this->connection = $connection;
     }
 
+    public function getConnection()
```

Comment:
Could you add a typehint for the return type? Why is this method needed?

Reply:
The modified` InvoicesDocument addInvoice` function uses the `InvoiceTypeApiConnector` and `ArticleApiConnector` to request additional information on the selected `InvoiceType` and `Articles` in the `Invoice`, which means it needs access to the current connection. The `InvoiceApiConnector sendAll` function calls the parent `BaseApiConnector getConnection` function and passes the connection as an argument to `addInvoice`.

Typehint added